cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/table/hbase/HBaseTable.java

```diff
@@ -296,6 +322,7 @@ private void setFilterIfNeeded(ScanBuilder scan, @Nullable Filter filter) {
     }
   }
 
+  // null columns and empty columns both mean to retrieve all columns
   private Get createGet(byte[] row, @Nullable byte[][] columns) {
```

Comment:
this should not allow empty cols. In the code that calls it, you only call if it is not empty. So I guess the comment above is inaccurate.